packages/preferences/src/browser/workspace-preference-provider.ts

```diff
@@ -16,13 +16,13 @@ export class WorkspacePreferenceProvider extends AbstractResourcePreferenceProvi
     @inject(WorkspaceService)
     protected readonly workspaceService: WorkspaceService;
 
-    async getUri(): Promise<URI> {
+    async getUri(): Promise<URI | undefined> {
         const root = await this.workspaceService.root;
         if (root) {
             const rootUri = new URI(root.uri);
             return rootUri.resolve('.theia').resolve('settings.json');
         }
-        return new Promise<URI>(() => { });
+        return undefined;
```

Comment:
reported upstream ?

Reply:
already merged

Reply:
ok so do you need to rebase current master branch with upstream master branch ?

Reply:
it would be nice, but later 
because I have to open another one PR for https://github.com/theia-demo-plugins/wiptheia/pull/30#issuecomment-396580887